### DISCUSSION TITLE
Export the API documentation to Word.

### DISCUSSION THREAD

**Discussion by shalousun** [TongchengOpenSource/smart-doc#363] [Ideas]
There is a significant demand from many community users for the functionality to export documentation to Word format. Currently, this feature is supported on Torna. However, smart-doc has not yet implemented this capability. The primary reason for this is that the ongoing maintenance of smart-doc is handled solely by @shalousun and @netdied. The original author of the project, @shalousun, is also no longer actively engaged in Java development, resulting in a severe shortage of manpower for the open-source maintenance of smart-doc.

The domestic user base of smart-doc is also quite large, and the issues and corresponding requirements are overwhelming to address in their spare time. We warmly welcome developers who are interested in joining the open-source community to collaborate and contribute to the development of smart-doc.

社区有很多同学有一些导出word的需求，目前在torna上已经支持word导出。
但是在smart-doc上一直没有该功能，主要原因是smart-doc当前的维护工作主要只有@shalousun和@netdied和两人。项目的作者@shalousun现在也很少再从事java的研发工作，smart-doc的开源维护工作非常缺乏人手。
smart-doc国内用户量也非常多，issue和对应的需求在工作之余已经无法应对，这里非常欢迎想加入开源社区的同学来一起共建smart-doc

# Development requirements for exporting to Word format in smart-doc.
- The provision of the Word template is subject to review by @shalousun（提供word的模板由@shalousun负责模板的审核）
- After the review is passed, the development of the Word XML data template will be undertaken.（审核通过之后，制作word的xml数据开发模板）
- 
Render the Word XML data template using the Beetl template engine and finally save it as a Word document（使用beetl模板引擎渲染word的xml数据模板，最后保存为word）

Students who are willing to provide a pull request (PR) can discuss here and submit their ideas.

**Comment by chenqi146** [comment]
[smart-doc-word.docx](https://github.com/TongchengOpenSource/smart-doc/files/13387457/smart-doc-word.docx)
word template

**Comment by chenqi146** [comment]
#686
